grpc-ballerina/auth_client_basic_auth_handler.bal

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/auth;
+
+# Represents credentials for Basic Auth authentication.
+public type CredentialsConfig record {|
+    *auth:CredentialsConfig;
+|};
+
+# Defines the Basic Auth handler for client authentication.
+public class ClientBasicAuthHandler {
+
+    auth:ClientBasicAuthProvider provider;
+
+    # Initializes the `http:ClientBasicAuthHandler` object.
+    #
+    # + config - The `http:CredentialsConfig` instance
+    public isolated function init(CredentialsConfig config) {
+        self.provider = new(config);
+    }
+
+    # Enrich the request with the relevant authentication requirements.
```

Comment:
IMO, this should be updated as enrich the headers, not the request.

Reply:
+1 fixed in https://github.com/ballerina-platform/module-ballerina-grpc/pull/74/commits/18553e442144a9bd0020f668b934292a29cf8465